airflow/models/taskinstance.py

```diff
@@ -773,7 +773,7 @@ def get_previous_start_date(
         """
         self.log.debug("previous_start_date was called")
         prev_ti = self.get_previous_ti(state=state, session=session)
-        return prev_ti and prev_ti.start_date
+        return prev_ti and prev_ti.start_date and pendulum.instance(prev_ti.start_date)
```

Comment:
Shouldn't this be:
```suggestion
        return prev_ti and pendulum.instance(prev_ti.start_date)
```

Reply:
@kaxil please see previous comment for reasining on this change.   the issue is in test cases there is no dag run --> no _actual_ ti so a new one is created _which has no start date_

to fix this, we'd have to add a dag run in these two virtualenv operator tests or something like that

Reply:
I would rather have:

```
return prev_ti and prev_ti.start_date is not None and pendulum.instance(prev_ti.start_date)
```
Although my brain is not working right now. Looping in @ashb @turbaszek @XD-DENG for reviews

Reply:
yes this or
```
if prev_ti and prev_ti.start_date:
    return pendulum.instance(prev_ti.start_date)
```

or 
```
if prev_ti and prev_ti.start_date is not None:
    return pendulum.instance(prev_ti.start_date)
```
All fine with me.  Will wait for guidance on changing test vs changing code.

Reply:
@kaxil i updated the python operator tests to create a dag run, so the operative code can assume presence of a `start_date`.